src/domain_features/APIs/TransactionsAPI/TransactionForm.jsx

```diff
@@ -1,39 +1,52 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Form } from 'formik';
-import { useNavigate } from 'react-router-dom';
-import { reset } from '../../../shared/context/redux/actions';
-import store from '../../../shared/context/redux/store';
-import { useGetCall, useLocationId, usePatchCall } from '../../../shared/custom_hooks/index';
+import { useDispatch, useSelector } from 'react-redux';
+import { useLocationId, useUpdateCallRedux, } from '../../../shared/custom_hooks/index';
 import { DrawerTitle, DrawerForm, FormButtons, SimpleCard } from "../../../shared/components/index";
 import { transactionFieldArray } from "../../../shared/components/form/userFormFieldsArray";
 import TransactionValidationSchema from '../../models/TransactionValidationSchema';
 import InputField from './FormInputField';
 import DB_URL from '../../../shared/utils/URLs';
+import {
+    selectedTransaction,
+    updateTransactionAction,
+    updateTransactionFailure,
+    updateTransactionSuccess,
+    removeSelectedTransaction
+} from '../../../shared/context/redux/actionCreators/transactions';
 
 export default function TransactionForm() {
-    const navigate = useNavigate();
+    const dispatch = useDispatch();
     const transactionId = useLocationId();
     const url = `${DB_URL}/transactions/${transactionId}`;
-    const { data: singleTransaction } = useGetCall(url, "fail_fetch_transaction");
-    const { updateEntry: updateTransaction } = usePatchCall(url, "fail_patch_transaction", "transactions");
+    const { updateEntry } = useUpdateCallRedux(url, "fail_update_transaction", "success_edit_transaction", "transactions", updateTransactionAction, updateTransactionSuccess, updateTransactionFailure);
+    const singleTransaction = useSelector((state) => state.allTransactions.transactions.find(transaction => transaction.id === transactionId));
+    const open = useSelector((state) => state.drawerReducer.open);
 
-    const cancelUpdateTransaction = () => {
-        navigate(-1);
-        store.dispatch(reset);
+    useEffect(() => {
+        if (transactionId && transactionId !== undefined) dispatch(selectedTransaction(singleTransaction));
+        return () => {
```

Comment:
do we need to have return () here or we could just dispatch the removeSelectedTransaction()